verification.md

```diff
@@ -11,24 +12,296 @@ module VERIFICATION
     imports EVM
 ```
 
-This `smt-lemma` helps Z3 reason about stack over/under flow.
+Abstractions
+------------
+
+We design abstractions that capture the EVM low-level specific details, allowing to specify specifications and reason about properties in a higher level similar to that of the surface languages (e.g., Solidity or Viper) in which smart contracts are written.
+
+### Memory Abstraction
+
+We present an abstraction for the EVM memory to allow the word-level reasoning. The word is considered as the smallest unit of values in the surface language level (thus in the contract developers’ mind as well), but the EVM memory is byte-addressable. Our abstraction helps to fill the gap and make the reasoning easier.
+
+Specifically, we introduce uninterpreted function abstractions and refinements for the word-level reasoning.
+
+The term `nthbyteof(v, i, n)` represents the i-th byte of the two's complement representation of v in n bytes (0 being MSB), with discarding high-order bytes when v is not fit in n bytes.
+
+```{.k .uiuck}
+  syntax Int ::= nthbyteof(Int, Int, Int) [function, smtlib(smt_nthbyteof)]
+```
+
+Precisely it is defined as follows.
+
+```{.k .uiuck}
+  rule nthbyteof(V, I, N) => nthbyteof(V /Int 256, I, N -Int 1) when N  >Int (I +Int 1) [concrete]
+  rule nthbyteof(V, I, N) =>           V %Int 256               when N ==Int (I +Int 1) [concrete]
+```
```

Comment:
Explain the concrete attribute.  Actually, when did you guys add it?  Wouldn't it be better to have a `#concrete` symbol and add a `requires` side condition saying that the (meta-)variables V, I, N are concrete?

Reply:
I thought that as well, but it seems more succinct than having the lengthy condition when we require all of the variables to be concrete.

Reply:
ok, acceptable.  but add a comment stating the relationship between these two notations when you use each of them for the first time.